uaclient/exceptions.py

```diff
@@ -306,6 +306,22 @@ def __str__(self):
         )
 
 
+class EntitlementNotEnabledError(UserFacingError):
+    """An exception raised when enabling of an entitlement fails"""
+
+    pass
+
+
+class FullAutoAttachFailureError(UserFacingError):
+    """An exception raised when auto attach at boot fails"""
+
+    def __init__(self):
+        super().__init__(
+            msg=messages.FULL_AUTO_ATTACH_ERROR.msg,
+            msg_code=messages.FULL_AUTO_ATTACH_ERROR.name,
+        )
```

Comment:
If these were added specifically to be raised by the api function, then I think we should go ahead and define them in `uaclient.api.exceptions` - thoughts @lucasmoura @CalvoM @renanrodrigo ?

Reply:
+1

Reply:
+1